pkg/kubectl/cmd/drain/drain.go

```diff
@@ -640,7 +430,7 @@ func (o *DrainOptions) deletePods(pods []corev1.Pod, getPodFn func(namespace, na
 	return err
 }
 
-func (o *DrainOptions) waitForDelete(pods []corev1.Pod, interval, timeout time.Duration, usingEviction bool, getPodFn func(string, string) (*corev1.Pod, error)) ([]corev1.Pod, error) {
+func (o *DrainCmdOptions) waitForDelete(pods []corev1.Pod, interval, timeout time.Duration, usingEviction bool, getPodFn func(string, string) (*corev1.Pod, error)) ([]corev1.Pod, error) {
```

Comment:
This function wants to write output a lot, I've not moved it cause I believe ideally libraries shouldn't write output. I've not figured out how alternative version of this would look like, and decide to not do that for the purpose of first PR.

Reply:
backends shouldn't be necessary on mute; they should allow piping to a custom stream.

i see this is already possible using `o.Out/ErrOut`, yet if this is always printing to the standard streams and if there is no easy way to customize it, it needs a minor refactor.
but given `o.Out/ErrOut` exists i'd consider moving this function too.

Reply:
Yes, but the streams are `genericclioptions.IOStreams`, and those are not convenient to use outside of Kubernetes project itself, as an additional wrapping is needed to attach those to a logger or something else.

Reply:
i see, so that's a problem, yes.
consumers should have an easy way to manage the streams.

Reply:
Yes, so I'm think of addressing this in a follow-up PR, what do you think?

Reply:
sounds good!